### PR TITLE
Resources type widget

### PR DESCRIPTION
**Pull request by rohitdobariyaleanspace**

### CODE REVIEW

**Review by coutinol — APPROVED** [review]

**Review by jerome-leanspace — CHANGES_REQUESTED** [review]
Also there some things that need to be changed
The metric used needs to have a unit and it's not present in this example could you fix it?
Also after applying when doing a second apply it seems it wants to update again

File: testing/dashboard/widgets/main.tf [hunk 1]
```diff
@@ -16,6 +16,11 @@ variable "numeric_metric_id" {
   description = "The ID of the numeric metric to create widgets for."
 }
 
+variable "resource_id" {
```
[the remark this answers is not in the document]
Comment (rohitdobariyaleanspace) [reply testing/dashboard/widgets/main.tf:19]: Nice catch, its need to be here its comes from the resource created in the resources module...

**Comment by jerome-leanspace** [comment]
> > Also there some things that need to be changed The metric used needs to have a unit and it's not present in this example could you fix it? Also after applying when doing a second apply it seems it wants to update again
> 
> You mean [here](https://github.com/leanspace/terraform-provider-leanspace/blob/f358c355794ccc3e1e2fd2f16bca761db1c0418d/examples/resources/leanspace_metrics.tf#L6), Unit is optional for numeric type.

Yes it's optional on a metric but it seems to be mandatory for a resource to be created, I don't know this service but this is an error that I got when running your terraform example

**Comment by rohitdobariyaleanspace** [comment]
> > > Also there some things that need to be changed The metric used needs to have a unit and it's not present in this example could you fix it? Also after applying when doing a second apply it seems it wants to update again
> > 
> > 
> > You mean [here](https://github.com/leanspace/terraform-provider-leanspace/blob/f358c355794ccc3e1e2fd2f16bca761db1c0418d/examples/resources/leanspace_metrics.tf#L6), Unit is optional for numeric type.
> 
> Yes it's optional on a metric but it seems to be mandatory for a resource to be created, I don't know this service but this is an error that I got when running your terraform example

Yes, you are right @jerome-leanspace, in the resources if create with metric (Numeric), unit must be exists in metric..., I am suprice how it works with resources terrafrom

I will check and resolve this...

**Review by jerome-leanspace — APPROVED** [review]

**Pull request merged**